src/Domain/Creators/TimeDependentOptions/Sphere.cpp

```diff
@@ -250,36 +316,35 @@ TimeDependentMapOptions::distorted_to_inertial_map(
 }
 
 TimeDependentMapOptions::MapType<Frame::Grid, Frame::Distorted>
-TimeDependentMapOptions::grid_to_distorted_map(
-    const bool include_distorted_map) const {
-  if (include_distorted_map) {
-    if (not shape_map_.has_value()) {
-      ERROR(
-          "Requesting grid to distorted map with distorted frame but shape map "
-          "options were not specified.");
-    }
-    return std::make_unique<GridToDistortedComposition>(shape_map_.value());
+TimeDependentMapOptions::grid_to_distorted_map(const size_t block_number,
+                                               const bool is_inner_cube) const {
+  const bool block_has_shape_map = shape_map_options_.has_value() and
+                                   block_number < (filled_ ? 12 : 6) and
+                                   not is_inner_cube;
+  if (block_has_shape_map) {
+    return std::make_unique<GridToDistortedComposition>(
+        gsl::at(shape_maps_, filled_ ? block_number : 0));
```

Comment:
Please add a comment here that when the inner cube isn't filled, then we use the sphere transition which is the same for all wedges and not the wedge transition. Hence, why we grab only the 0th shape map regardless of block. Also below in the grid to inertial map